aito/sdk/aito_client.py

```diff
@@ -0,0 +1,565 @@
+        :type polling_time: int
+        :raises RequestError: an error occurred during the upload of the file to S3
+        """
+        LOG.debug(f'uploading file object to table `{table_name}`...')
+        LOG.debug('initiating file upload...')
+        r = self.request('POST', f"/api/v1/data/{table_name}/file")
+        LOG.info('file upload initiated')
+        LOG.debug('getting session id and upload url...')
+        upload_session_id = r['id']
+        session_end_point = f'/api/v1/data/{table_name}/file/{upload_session_id}'
+        s3_url = r['url']
+        upload_req_method = r['method']
+        LOG.info('got session id and upload url')
+        LOG.debug('uploading file file to s3...')
+        try:
+            r = requests.request(upload_req_method, s3_url, data=binary_file)
+            r.raise_for_status()
+        except Exception as e:
+            raise BaseError(f'fail to upload file to S3: {e}')
+        LOG.debug('uploaded file to S3')
+        LOG.debug('triggering file processing...')
+        self.request('POST', session_end_point)
+        LOG.info('triggered file processing')
+        LOG.debug('polling processing status...')
+        while True:
+            try:
+                processing_progress = self.request('GET', session_end_point)
+                status = processing_progress['status']
+                LOG.debug(f"completed count: {status['completedCount']}, throughput: {status['throughput']}")
+                if processing_progress['errors']['message'] != 'Last 0 failing rows':
+                    LOG.error(processing_progress['errors'])
+                if status['finished']:
+                    break
+            except Exception as e:
+                LOG.debug(f'failed to get file upload status: {e}')
+            time.sleep(polling_time)
+        LOG.info(f'uploaded file object to table `{table_name}`')
+        if optimize_on_finished:
+            self.optimize_table(table_name)
+
+    def upload_file(
+            self, table_name: str, file_path: PathLike, optimize_on_finished: bool = True, polling_time: int = 10):
+        """upload a file to a table `API doc <https://aito.ai/docs/api/#post-api-v1-data-table-file>`__
+
+        :param table_name: the name of the table
+        :type table_name: str
+        :param file_path: path to uploading file
+        :type file_path: PathLike
+        :param optimize_on_finished: `optimize https://aito.ai/docs/api/#post-api-v1-data-table-optimize`__ the table on finished, default to True
+        :type optimize_on_finished: bool
+        :param polling_time: polling wait time
+        :type polling_time: int
+        :raises BaseError: the uploading file is not a ndjson.gz file
+        :raises RequestError: an error occurred during the upload of the file to S3
+        """
+        if not check_file_is_gzipped(file_path):
+            raise BaseError(f'uploading file {file_path} is not a gzip compressed ndjson file')
```

Comment:
```suggestion
            raise BaseError('file {file_path} is not a gzip-compressed ndjson file')
```
I think it is clear from the context that the file mentioned is the one chosen to be uploaded.